sdk/identity/azure-identity/azure/identity/_credentials/silent.py

```diff
@@ -57,47 +60,69 @@ def get_token(self, *scopes: str, **kwargs: Any) -> AccessToken:
         if not scopes:
             raise ValueError('"get_token" requires at least one scope')
 
-        if not self._initialized:
-            self._initialize()
+        token_cache = self._cae_cache if kwargs.get("enable_cae") else self._cache
+
+        # Try to load the cache if it is None.
+        if not token_cache:
+            token_cache = self._initialize_cache(is_cae=bool(kwargs.get("enable_cae")))
 
-        if not self._cache:
-            if self._is_chained:
-                raise CredentialUnavailableError(message="Shared token cache unavailable")
-            raise ClientAuthenticationError(message="Shared token cache unavailable")
+            # If the cache is still None, raise an error.
+            if not token_cache:
+                if self._is_chained:
+                    raise CredentialUnavailableError(message="Shared token cache unavailable")
+                raise ClientAuthenticationError(message="Shared token cache unavailable")
 
         return self._acquire_token_silent(*scopes, **kwargs)
 
-    def _initialize(self):
-        if not self._cache and platform.system() in {"Darwin", "Linux", "Windows"}:
+    def _initialize_cache(self, is_cae: bool = False) -> Optional[TokenCache]:
+
+        # If no cache options were provided, the default cache will be used. This credential accepts the
+        # user's default cache regardless of whether it's encrypted. It doesn't create a new cache. If the
+        # default cache exists, the user must have created it earlier. If it's unencrypted, the user must
+        # have allowed that.
+        cache_options = self._cache_persistence_options or TokenCachePersistenceOptions(allow_unencrypted_storage=True)
+
+        if platform.system() not in {"Darwin", "Linux", "Windows"}:
+            raise CredentialUnavailableError(message="Shared token cache is not supported on this platform.")
+
+        if not self._cache and not is_cae:
             try:
-                # If no cache options were provided, the default cache will be used. This credential accepts the
-                # user's default cache regardless of whether it's encrypted. It doesn't create a new cache. If the
-                # default cache exists, the user must have created it earlier. If it's unencrypted, the user must
-                # have allowed that.
-                options = self._cache_persistence_options or TokenCachePersistenceOptions(
-                    allow_unencrypted_storage=True
-                )
-                self._cache = _load_persistent_cache(options)
+                self._cache = _load_persistent_cache(cache_options, cache_suffix=CACHE_NON_CAE_SUFFIX)
```

Comment:
I would suggest making a helper function to return the name.

Then we don't need to handle CACHE_NON_CAE_SUFFIX, CACHE_CAE_SUFFIX everywhere.

Reply:
Hmm, I can move the suffix handling inside the _load_persistent_cache function to avoid needing to import/use it everywhere.